### ISSUE TITLE
Terraform Provider Development Program - Initial Review

### ISSUE THREAD

**Issue by cgriggs01** [hedvig/terraform-provider-hedvig#1]
Hi There,

I’ve taken a look at the provider and would like to say it looks good so far. I do have feedback outlined below that I’d like to see addressed before we move on to the next steps.

- [x] Repository Name: Can you change the name of this repository to terraform-provider-hedvig/. This is because Go default naming convention when building is the root directory of the provider. As well, once the provider review process is complete, this repository will be ingested to the [Terraform Providers](github.com/terraform-providers/) Github organization where the name is necessary for the naming convention. 

- [x] Missing READ and UPDATE functions in resource files.  Many of your Resources READ and UPDATE methods only return `nil' without performing, these will need to be added. Also, as a best practice, the CREATE method of any Resource should at least return the results of it’s own READ method. In certain cases it may need to return the results of the UPDATE method, depending on if the CREATE method is capable of setting all possible configuration values. If for example the API call to create Resource only allows basic creation, the remaining configuration is applied via update, then CREATE should return UPDATE, and UPDATE should return READ. AWS Security Group Resource is an example of this; the CREATE only allows creating a Security Group with a name and description, and we must make other API calls to apply the rules. Thus, CREATE returns the results of UPDATE.

     UPDATE methods should always return the results of the READ method.

- [x] File Format: The Each provider has a section in the Terraform documentation. You'll want to add new index file and individual pages for each resource supported by the provider.A good example of the standard provider file structure can be found in the [AWS provider](https://github.com/terraform-providers/terraform-provider-aws)

- [x] Documentation: Each provider has a section in the Terraform documentation. You'll want to add a new index file and individual pages for each resource supported by the provider. A good example of a simple set of website docs can be found at [terraform-provider-template/](GitHub.com/terraform-providers/terraform-provider-template/).

Please let me know if you have any questions.

Best,
Chris

**Comment by katrinahayes** [comment]
Hello Chris,
Thank you for your feedback. We have renamed the repository. We are working on implementing the read and update methods and filling out the documentation, and will let you know when those are ready.

Thanks again,
Katrina

**Comment by katrinahayes** [comment]
Hello Chris,
We have completed the READ and UPDATE functions in the resource files and updated the other functions to return the READ methods instead of nils. We have also updated our provider’s file structure and filled in the website documentation to follow the examples provided by other providers.

We look forward to hearing your feedback.

Thank you,
Katrina

**Comment by cgriggs01** [comment]
Hi Katrina,

Thank you for making those changes, I will make another pass. Is there a reason for keeping the work you have done so far in the 'dev' branch?

Before we can start running the acceptance tests,  your repository will need a [GNUmakefile](https://github.com/terraform-providers/terraform-provider-aws/blob/master/GNUmakefile) and corresponding [scripts/](https://github.com/terraform-providers/terraform-provider-aws/tree/master/scripts) directory and files. This is so we are able to run 'make testacc' to run the providers acceptance tests. 

You will also need a 1.0.0 [CHANGELOG.md](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md) that will be used to tag binary releases of the provider. 

Best,
Chris

**Comment by katrinahayes** [comment]
Hello Chris,
I have added the necessary files to the repository; they are available at https://github.com/hedvig/terraform-provider-hedvig/tree/dev. We just keep everything in a ‘dev’ branch by convention, but can merge everything into ‘master’ or another branch if that is preferable.

We will be adding the changelog shortly.

Thank you!
Katrina

**Comment by cgriggs01** [comment]
Great, I look forward to testing again. 

Yes - it would be preferred if your you keep your stable provider under the 'master' branch. This helps to minimize confusion throughout the review process. 

Could you add two more files to your [scripts/](https://github.com/hedvig/terraform-provider-hedvig/tree/dev/scripts) directory? The files, [changelog-links.sh](https://github.com/terraform-providers/terraform-provider-aws/blob/master/scripts/changelog-links.sh) and [gogetcookie.sh](https://github.com/terraform-providers/terraform-provider-aws/blob/master/scripts/gogetcookie.sh) are used during the deployment process by our CI release process. 

A hedvig.erb file need to be added to your [website/](https://github.com/hedvig/terraform-provider-hedvig/tree/dev/website) directory as a Ruby layout file for the providers terraform.io website documentation. A good example of this file can be found in the [Template provider](https://github.com/terraform-providers/terraform-provider-template/blob/master/website/template.erb). Then move the index file and resource directory into a docs/ folder. The website file structure should then look as follows: 
```
website/
     hedvig.erb
     docs/
          index.html.markdown
          r/
```

Any vendor code that is used by your provider should be included in a `vendor/` directory.

If you have any questions please let me know. 

Thank you for your time and effort,
Chris

**Comment by katrinahayes** [comment]
Hello Chris,
I have merged the dev branch into the master branch, so the stable provider is now available in master. I have also added the two scripts to the scripts/ directory. I will add the changelog, make the changes to the website directory and add the hedvig.erb file early next week.

Thank you,
Katrina

**Comment by katrinahayes** [comment]
Hello Chris,
The changelog and the updates to the website are now available in our master branch.

Thank you,
Katrina

**Comment by cgriggs01** [comment]
Hi Katrina,
I hope things are well. The provider is looking good! I just have a few more items to address. 

- [ ] The Changelog's next version should be listed at the top with "(Unreleased)" following the version number. i.e. `1.0.0 (Unreleased)` This is necessary for our semi-automated release process. 

- [ ] We recently made an addition to the standard makefile, can you add the `website` and `website-test` targets to the GNUmakefile. This PR shows the exact changed needed - [Provider PR](https://github.com/terraform-providers/terraform-provider-scaffolding/pull/2/files)

- [ ] It's a standard for all Terraform provider to all have a [MPL2 License](https://github.com/terraform-providers/terraform-provider-scaffolding/blob/master/LICENSE), would you be able to add this to the repository?

- [ ] All of the providers dependancies should be checked into the `vendor/` directory pushed upto the repository 

The acceptance test are all passing, I just want to verify something about the Hedvig environment. Is the testing environment behind a VPN because you are hosting it, or is it because Hedvig envs are always behind a VPN. We probably discussed this awhile back but I cannot remember now. 

Looking forward to releasing this provider!

Best,
Chris

**Comment by katrinahayes** [comment]
Hello Chris,
Great to hear from you! I have addressed all the items you mentioned; they are committed and available now in the master branch at https://github.com/hedvig/terraform-provider-hedvig/. Regarding your question about using a VPN, it is only required because we are hosting the test cluster. Most users will be able to use the Hedvig Provider without a VPN.

Thank you!

Katrina

**Comment by cgriggs01** [comment]
Hi Katrina, 

I will be closing this issue now, and my teammate will be opening a new issue for their review. 

Best,
Chris
